src/error.rs

```diff
@@ -0,0 +1,59 @@
+use std::fmt;
+
+#[derive(Debug)]
+/// A union of all possible errors in our lib
+pub enum KvStoreError {
```

Comment:
Can be more concise:
```suggestion
pub enum KvsError {
```